src/main/groovy/io/nextflow/gradle/registry/RegistryClient.groovy

```diff
@@ -33,14 +38,31 @@ class RegistryClient {
              def rep = http.execute(req)) {
 
             if (rep.statusLine.statusCode != 200) {
-                def err = gson.fromJson(new InputStreamReader(rep.entity.content), ErrorResponse)
-                throw new RuntimeException("Failed to publish plugin: ${err.type} - ${err.message}")
+                throw new RuntimeException(getErrorMessage(rep))
```

Comment:
Not sure how much it's worth mapping the response to the ErrorResponse, there are entire classes of non-200 status that are not errors (2xx and 3xx) also 5xx error could have their own error format. 

I would create a custom exception adding the response text as an additional details in the error message as it is (without paring it)